docs/src/docs/asciidoc/configuration-model.adoc

```diff
@@ -3,22 +3,231 @@
 :toc: left
 :toclevels: 1
 
-[[oauth2-authorization-server-configurer]]
-== OAuth2AuthorizationServerConfigurer
+[[configure-defaults]]
+== Default configuration
```

Comment:
You've described best practice in the technical writing and publishing communities: lower-case everything and change spaces to hyphens. Problematic characters for URLs (such as ampersands) get dropped entirely.

So, yes, a heading of `Default Configuration` gets an anchor of `default-configuration`.

Adding the document name is often unnecessary, but I think it will help in this guide, because you will end up with the same headings in several places (as common children of the higher headings). So, in this case, it's a good convention to adopt. Note that, once we switch to Antora, Antora fixes the problem for us (by programmatically adding an identifier when it finds a duplicate heading in two source files). But it'll work both before and after we get to Antora, so do it to it. :)

Reply:
I updated to match names for heading and anchor. I'm not sure about prefixing the document name but I'll leave this to you @sjohnr and if you think it's best please go ahead in a separate commit.